src/TheThingsNetwork.cpp

```diff
@@ -612,18 +650,16 @@ void TheThingsNetwork::showStatus()
   debugPrintIndex(SHOW_RX_DELAY_2, buffer);
 }
 
-void TheThingsNetwork::configureEU868()
+// Puting this common fonction saves 238 bytes of flash
+void TheThingsNetwork::configureChannelsFreq(uint32_t freq, uint8_t first, uint8_t last, uint8_t first_dr)
 {
-  sendMacSet(MAC_RX2, "3 869525000");
-  sendChSet(MAC_CHANNEL_DRRANGE, 1, "0 6");
-
-  char buf[10];
-  uint32_t freq = 867100000;
   uint8_t ch;
-  for (ch = 0; ch < 8; ch++)
+  char buf[10];
```

Comment:
Although I know that `sprintf()` doesn't allocate memory, all these small buffers on the stack do allocate extra memory. So yeah the code is smaller, and we'll leave it this way, but do note that the end result is that this code requires more (dynamic) memory.

Reply:
I really don't like using `sprintf()` on embedded devices, unless it is really necessary. I also prefer to use statically allocated memory, rather than many dynamically allocated variables. Using statically allocated ones you can know at compile time that you will run out of memory. Doing it dynamically will just cause strange behaviour down the line.

Reply:
Guys,
just a a reminder, sprintf stuff has nothing to do with my PR and was present is the code before I made any changes. It was present and used, so I used it also for the same things it was already used and thought it was consistent even if I agree about dynamically allocated variables.
Charles

Reply:
Sprintf  wasnt implemented in this PR, so that would be stuff for an issue @jpmeijers

Reply:
@hallard @samuel-puschacher you are changing existing code and replacing that by using `sprintf()`. That has nothing to do with this PR.